### PR TITLE
yuzu: Change QtKeyToSwitchKey switch case to array

### PR DESCRIPTION
**Pull request by german77**
Replaces ugly code with compact ugly code

### CODE REVIEW

File: src/yuzu/bootmanager.cpp [hunk 1]
```diff
@@ -401,224 +401,127 @@ void GRenderWindow::closeEvent(QCloseEvent* event) {
...
-    case Qt::Key_Katakana:
-        return Settings::NativeKeyboard::KatakanaKey;
-    case Qt::Key_Hiragana:
-        return Settings::NativeKeyboard::HiraganaKey;
-    case Qt::Key_Zenkaku_Hankaku:
-        return Settings::NativeKeyboard::ZenkakuHankaku;
-    // Modifier keys are handled by the modifier property
-    default:
-        return Settings::NativeKeyboard::None;
+    constexpr std::array<std::pair<Qt::Key, Settings::NativeKeyboard::Keys>, 106> key_map = {
```
Comment (lioncash) [inline src/yuzu/bootmanager.cpp:404]:
```suggestion
    static constexpr std::array<std::pair<Qt::Key, Settings::NativeKeyboard::Keys>, 106> key_map = {
```

File: src/yuzu/bootmanager.cpp [hunk 2]
```diff
@@ -401,224 +401,127 @@ void GRenderWindow::closeEvent(QCloseEvent* event) {
...
+        // {Qt::..., Settings::NativeKeyboard::NumPadCommaPc98},
+        {Qt::Key_Hangul, Settings::NativeKeyboard::Muhenkan},
+        {Qt::Key_Hangul_Hanja, Settings::NativeKeyboard::HangulEnglish},
+        {Qt::Key_Katakana, Settings::NativeKeyboard::Hanja},
+        {Qt::Key_Hiragana, Settings::NativeKeyboard::KatakanaKey},
+        {Qt::Key_Zenkaku_Hankaku, Settings::NativeKeyboard::ZenkakuHankaku},
+        // Modifier keys are handled by the modifier property
+    };
+
+    for (const auto [qkey, nkey] : key_map) {
```
Comment (lioncash) [inline src/yuzu/bootmanager.cpp:518]:
```suggestion
    for (const auto& [qkey, nkey] : key_map) {
```

File: src/yuzu/bootmanager.cpp [hunk 3]
```diff
@@ -401,224 +401,127 @@ void GRenderWindow::closeEvent(QCloseEvent* event) {
...
+        // {Qt::..., Settings::NativeKeyboard::Ro},
+        {Qt::Key_Hiragana_Katakana, Settings::NativeKeyboard::KatakanaHiragana},
+        {Qt::Key_yen, Settings::NativeKeyboard::Yen},
+        {Qt::Key_Henkan, Settings::NativeKeyboard::Henkan},
+        {Qt::Key_Muhenkan, Settings::NativeKeyboard::Muhenkan},
+        // {Qt::..., Settings::NativeKeyboard::NumPadCommaPc98},
+        {Qt::Key_Hangul, Settings::NativeKeyboard::Muhenkan},
+        {Qt::Key_Hangul_Hanja, Settings::NativeKeyboard::HangulEnglish},
+        {Qt::Key_Katakana, Settings::NativeKeyboard::Hanja},
+        {Qt::Key_Hiragana, Settings::NativeKeyboard::KatakanaKey},
```
Comment (Morph1984) [inline src/yuzu/bootmanager.cpp:513]: These seem to be mixed up

Comment (german77) [reply src/yuzu/bootmanager.cpp:513]: Indeed. I will fix them soon

**Review by liamwhite — APPROVED** [review]

**Review by bunnei — APPROVED** [review]

**Pull request merged**